microsoft/testsuites/power/power.py

```diff
@@ -0,0 +1,680 @@
+            log.info(
+                "Verified VM status is stopped after hibernation command sent")
+        else:
+            raise LisaException(
+                "Failed to change VM status to be stopped"
+                " after hibernation command sent")
+
+        # Resume the VM
+        start_stop.start()
+        log.info(
+            "Waked up the VM and continue checking its status"
+            " in every 15 seconds until 15 minutes timeout")
+
+        # Wait for VM resume for 15 min-timeout
+        timeout = 900
+        vm_resumed = False
+        start_time = time.time()
+        while (time.time() - start_time < timeout):
+            time.sleep(15)
+            is_ready, tcp_error_code = wait_tcp_port_ready(
+                node.public_address,
+                node.public_port,
+                log=log,
+                timeout=30,
+            )
+            if is_ready:
+                cmd_res = node.execute(
+                    "dmesg | grep -i 'hibernation exit'",
+                    shell=True,
+                    sudo=True
+                )
+                # TODO: THIS DMESG CHECKING APPROACH MAY NEED IMPROVE
+                res = re.search(r'(.*?hibernation exit.*?)',
+                                dmesg.get_output(force_run=True), re.IGNORECASE)
+                if not res or "hibernation exit" not in res.group(1):
+                    log.error(
+                        "VM resumed successfully "
+                        "but could not determine hibernation completion")
+                else:
+                    log.info("VM resumed successfully")
+                    vm_resumed = True
+                    break
+            else:
+                log.debug("VM is still resuming!")
+
+        if vm_resumed:
+            log.info("VM resume completed.")
+        else:
+            raise LisaException("VM resume did not finish.")
+
+        # Note that if you use NTP, the hardware clock is
+        # automatically synchronized to the system clock every 11 minutes,
+        # and this command is useful only at boot time
+        # to get a reasonable initial system time.
+        log.info("Waiting for RTC re-sync in 12 minutes...")
+        time.sleep(720)
+
+        log.info("Capturing the RTC timestmap")
+        after_time_stamp = _get_year(node)
+        log.debug(f"Got after_time_stamp: {after_time_stamp}")
+
+        # Verify the kernel panic, call trace or fatal error
+        res = re.findall('(call trace|fatal error)',
+                         dmesg.get_output(force_run=True), re.IGNORECASE)
+        if res:
+            log.error("Found Call Trace or Fatal error in dmesg")
+            # This is linux-next,
+            # so there is high chance to get call trace from other issue.
+            # For now, only print the error.
+        else:
+            log.info("Not found Call Trace and Fatal error in dmesg")
+        log.info("Waiting 60-second for logging sync")
+        time.sleep(60)
```

Comment:
not to hard code sleep.

Reply:
changed to a loop check and timeout in 60s